geomstats/geometry/discrete_curves.py

```diff
@@ -1314,6 +1314,9 @@ def discrete_horizontal_geodesic(
         if initial_point.ndim != end_point.ndim:
             initial_point, end_point = gs.broadcast_arrays(initial_point, end_point)
 
+        if callable(end_spline):
```

Comment:
So the issue is that the `check_is_batch` function in `align` is only considering `point`, i.e. for the use case 1 `point`, multiple `base_point` it creates only one spline instead of a list, which breaks the `for` within `discrete_horizontal_geodesic`.

I could have fixed it in `align`, but I thought it is not worthy to create multiple equal splines. Therefore, with the `callable` check we ensure the code works properly for the use case I mentioned above.

Maybe to make the code more readable we can change `align` and keep `callable`

```python
if point.ndim > bundle.total_space.point_ndim:
  spline = bundle.total_space.interpolate(point)
else:
  spline = [bundle.total_space.interpolate(point_) for point_ in point]
```

or change `align` and remove `callable`

```python
if point.ndim > bundle.total_space.point_ndim`:
  spline = bundle.total_space.interpolate(point)
  if base_point.ndim > bundle.total_space.point_ndim:
    spline = [spline]*base_point.shape[0]
else:
  spline = [bundle.total_space.interpolate(point_) for point_ in point]
```

Which solution do you prefer?

Reply:
Ah I see ! I think I like the second solution better because it's easier to understand when reading the code.